apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/actions.ts

```diff
@@ -1,53 +1,67 @@
 "use server";
 
-import { getServerSession } from "next-auth";
-import { authOptions } from "@formbricks/lib/authOptions";
+import { z } from "zod";
+import { authenticatedActionClient } from "@formbricks/lib/actionClient";
+import { checkAuthorization } from "@formbricks/lib/actionClient/utils";
+import { getOrganizationIdFromSurveyId } from "@formbricks/lib/organization/utils";
 import { getResponseDownloadUrl, getResponseFilteringValues } from "@formbricks/lib/response/service";
-import { canUserAccessSurvey, verifyUserRoleAccess } from "@formbricks/lib/survey/auth";
 import { updateSurvey } from "@formbricks/lib/survey/service";
 import { getTagsByEnvironmentId } from "@formbricks/lib/tag/service";
-import { AuthorizationError } from "@formbricks/types/errors";
-import { TResponseFilterCriteria } from "@formbricks/types/responses";
-import { TSurvey } from "@formbricks/types/surveys/types";
+import { ZResponseFilterCriteria } from "@formbricks/types/responses";
+import { ZSurvey } from "@formbricks/types/surveys/types";
 
-export const getResponsesDownloadUrlAction = async (
-  surveyId: string,
-  format: "csv" | "xlsx",
-  filterCritera: TResponseFilterCriteria
-): Promise<string> => {
-  const session = await getServerSession(authOptions);
-  if (!session) throw new AuthorizationError("Not authorized");
+const ZGetResponsesDownloadUrlAction = z.object({
+  surveyId: z.string(),
+  format: z.union([z.literal("csv"), z.literal("xlsx")]),
+  filterCriteria: ZResponseFilterCriteria,
+});
 
-  const isAuthorized = await canUserAccessSurvey(session.user.id, surveyId);
-  if (!isAuthorized) throw new AuthorizationError("Not authorized");
+export const getResponsesDownloadUrlAction = authenticatedActionClient
+  .schema(ZGetResponsesDownloadUrlAction)
+  .action(async ({ ctx, parsedInput }) => {
+    await checkAuthorization({
+      userId: ctx.user.id,
+      organizationId: await getOrganizationIdFromSurveyId(parsedInput.surveyId),
+      rules: ["survey", "read"],
+    });
 
-  return getResponseDownloadUrl(surveyId, format, filterCritera);
-};
+    return getResponseDownloadUrl(parsedInput.surveyId, parsedInput.format, parsedInput.filterCriteria);
+  });
 
-export const getSurveyFilterDataAction = async (surveyId: string, environmentId: string) => {
-  const session = await getServerSession(authOptions);
-  if (!session) throw new AuthorizationError("Not authorized");
+const ZGetSurveyFilterDataAction = z.object({
+  surveyId: z.string(),
+  environmentId: z.string(),
```

Comment:
Can we derive the `environmentId` from the `survey` to make sure we don't have inconsistencies?